src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -0,0 +1,176 @@
+//
+// - '1'
+// - [{NodeID:1,Core:0},{NodeID:2,Core:0},{NodeID:3,Core:0}].
+func parsePartitionReplicas(partition string) (string, []adminapi.Replica, error) {
+	var replicas []adminapi.Replica
+	pr := strings.Split(partition, ":")
+	if len(pr) != 2 {
+		return "", nil, fmt.Errorf("invalid format for %s", partition)
+	}
+	if strings.Contains(pr[0], "/") {
+		return "", nil, fmt.Errorf("invalid format for %s; unexpected \"/\"", partition)
+	}
+	replicasTmp := strings.Split(pr[1], ",")
+	for _, replicaCore := range replicasTmp {
+		nc := strings.Split(replicaCore, "-")
+		if len(nc) > 1 {
+			nodeID, err := strconv.Atoi(nc[0])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			core, err := strconv.Atoi(nc[1])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[1], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+				Core:   core,
+			})
+		} else {
+			nodeID, err := strconv.Atoi(replicaCore)
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+			})
+		}
+	}
+	return pr[0], replicas, nil
+}
+
+// Expected input format is 'foo/0:1,2,3' or 'redpanda/controller/0:1-0,2-0,3-0'
+// If 'foo/0:1,2,3' is passed, returns
+//
+// - 'kafka/foo/0'
+// - [{NodeID:1,},{NodeID:2},{NodeID:3}]
+//
+// If 'redpanda/controller/0:1-0,2-0,3-0' is passed, returns
+//
+// - 'redpanda/controller/0'
+// - [{NodeID:1,Core:0},{NodeID:2,Core:0},{NodeID:3,Core:0}].
+func parseTopicPartitionReplicas(partition string) (string, []adminapi.Replica, error) {
+	var replicas []adminapi.Replica
+	tpr := strings.Split(partition, ":")
+	if len(tpr) != 2 {
+		return "", nil, fmt.Errorf("invalid format for %s", partition)
+	}
+	if !strings.Contains(tpr[0], "/") {
+		return "", nil, fmt.Errorf("invalid format for %s; specify the topic name", tpr[0])
+	}
+	if strings.Count(tpr[0], "/") == 1 {
+		tpr[0] = "kafka/" + tpr[0]
+	}
+	replicasTmp := strings.Split(tpr[1], ",")
+	for _, replicaCore := range replicasTmp {
+		nc := strings.Split(replicaCore, "-")
+		if len(nc) > 1 {
+			nodeID, err := strconv.Atoi(nc[0])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			core, err := strconv.Atoi(nc[1])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[1], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+				Core:   core,
+			})
+		} else {
+			nodeID, err := strconv.Atoi(replicaCore)
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+			})
+		}
+	}
+	return tpr[0], replicas, nil
+}
+
+const helpAlterAssignments = `Move partition replicas across nodes / cores".
```

Comment:
There is an extra quote here
```suggestion
const helpAlterAssignments = `Move partition replicas across nodes / cores.
```

Reply:
Fixed in https://github.com/redpanda-data/redpanda/commit/1595272ed458ac47621cdbcad4c550c99140d622